### DISCUSSION TITLE
Bevy frame-dependent rendering

### DISCUSSION THREAD

**Discussion by torsteingrindvik** [bevyengine/bevy#11809] [Q&A] [answered]
I have a use case where instead of using time deltas to calculate transforms smoothly across frames, I want to instead be able to render my program determinstically given some frame number.

So then I can have some resource `MyFrameCount(u64)` or something, and calculate transforms from that. I don't expect many issues here.

But I also have animations, and as far as I can tell there doesn't seem to be a well-fitting API or approach to having them be determinstic.

Perhaps I can have animations paused always, then do something like

```rust
let anim_seek = (my_frame_count * anim_speed) % clip_duration; // since seek API must be in clip duration range
```

but maybe someone has any advice of what to do/what to expect?

**Comment by mockersf** [comment, marked as answer]
you can use the `Time` resource everywhere, and add the resource [`TimeUpdateStrategy::ManualDuration`](https://docs.rs/bevy/latest/bevy/time/enum.TimeUpdateStrategy.html#variant.ManualDuration).

That way you don't have to have special code for test!

**Reply by torsteingrindvik** [reply]
Thanks, the `ManualInstant` variant along with tracking my own starting instant + an offset based on simulation frame count works wonders.

Only problem is that I can only step forwards because of a limitation in the Bevy time impl, but I can make an issue for that.
